.github/workflows/docker.yml

```diff
@@ -0,0 +1,282 @@
+
+            - name: Set up QEMU
+              uses: docker/setup-qemu-action@v1
+            - name: Set up Docker Buildx
+              uses: docker/setup-buildx-action@v1
+
+            # Login to registry
+
+            - name: Log in to the Container registry
+              uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+              with:
+                  registry: ${{ env.REGISTRY }}
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            # Metadata and cache
+            - name: Extract metadata (tags, labels) for Docker database
+              id: meta-database
+              uses: docker/metadata-action@v3
+              with:
+                  images: ${{ env.REGISTRY }}/${{ github.repository }}/database
+
+            # Build and push
+
+            - name: Build and push Docker image
+              uses: docker/build-push-action@v2
+              with:
+                  context: .
+                  push: true
+                  tags: ${{ steps.meta-database.outputs.tags }}
+                  labels: ${{ steps.meta-database.outputs.labels }}
+                  file: ./docker/database.Dockerfile
+
+    build-darkflame-171023:
+        name: "Build Docker Image: darkflame-171023"
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            packages: write
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v2
+              with:
+                  submodules: recursive
+
+            - name: Set up QEMU
+              uses: docker/setup-qemu-action@v1
+            - name: Set up Docker Buildx
+              uses: docker/setup-buildx-action@v1
+
+            # Login to registry
+
+            - name: Log in to the Container registry
+              uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+              with:
+                  registry: ${{ env.REGISTRY }}
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            # Metadata and cache
+            - name: Extract metadata (tags, labels) for Docker darkflame-171023
+              id: meta-darkflame-171023
+              uses: docker/metadata-action@v3
+              with:
+                  images: ${{ env.REGISTRY }}/${{ github.repository }}/darkflame-171023
+
+            # Build and push
+
+            - name: Build and push Docker image
+              uses: docker/build-push-action@v2
+              with:
+                  context: .
+                  push: true
+                  tags: ${{ steps.meta-darkflame-171023.outputs.tags }}
+                  labels: ${{ steps.meta-darkflame-171023.outputs.labels }}
+                  file: ./docker/Dockerfile
+                  build-args: |
+                      BUILD_VERSION=171023
+                      BUILD_THREADS=2
+
+    build-darkflame-171022:
```

Comment:
If the only difference here is 171022 vs 171023, would it then maybe be possible to use a [build matrix](https://docs.github.com/en/actions/using-jobs/using-a-build-matrix-for-your-jobs) to only have to define this job once?